vmdb/app/models/ems_openstack_infra.rb

```diff
@@ -1,8 +1,24 @@
 class EmsOpenstackInfra < EmsInfra
   include EmsOpenstackMixin
 
+  before_create :ensure_parent_provider
```

Comment:
right, I was thinking if changes will be needed.

But I guess when I added zone, the zone can change.